### PR TITLE
[simple] Transaction generator improvements

### PR DESCRIPTION
**Pull request by igor-aptos**
### Description

- don't fail if only stale seq_num are received from rest endpoint
- workflow:
  - make delay_between_stages_s configurable
  - make init_max_gas_per_txn cli argument
  - ReducedBatchWrapperTransactionGeneratorCreator utility, useful for tournament

### CODE REVIEW

**Comment by igor-aptos** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #11863** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11863?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #11384** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11384?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #11481** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11481?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #11463** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11463?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #11465** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11465?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #11436** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11436?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/11863?utm_source=stack-comment).

File: crates/transaction-generator-lib/src/reduced_batch_wrapper.rs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+impl TransactionGenerator for ReducedBatchWrapperTransactionGenerator {
+    fn generate_transactions(
+        &mut self,
+        account: &LocalAccount,
+        num_to_create: usize,
+    ) -> Vec<SignedTransaction> {
+        self.generator
+            .generate_transactions(account, num_to_create.min(self.batch_size))
```
Comment (manudhundi) [inline crates/transaction-generator-lib/src/reduced_batch_wrapper.rs:19]: nit: So technically this is not always reduced. It is just capped/limited/bounded. Is "BoundedBatchWrapperTransactionGenerator" a better name ?

**Review by manudhundi — COMMENTED** [review]
Looks ok, except for one minor nit

**Review by sitalkedia — APPROVED** [review]
LGTM

File: crates/transaction-emitter-lib/src/cluster.rs [hunk 2]
```diff
@@ -96,9 +96,10 @@ impl Cluster {
             let state_timestamp = state.timestamp_usecs / 1000000;
             if state.chain_id != chain_id.id() {
                 warn!(
-                    "Excluding client {} running wrong chain {}",
+                    "Excluding client {} running wrong chain {}, instead of {}",
```
Comment (sitalkedia) [inline crates/transaction-emitter-lib/src/cluster.rs:99]: Just curious, when does this happen?

File: crates/transaction-emitter-lib/src/args.rs [hunk 3]
```diff
@@ -164,6 +164,9 @@ pub struct EmitArgs {
     #[clap(long)]
     pub max_gas_per_txn: Option<u64>,
 
+    #[clap(long)]
+    pub init_max_gas_per_txn: Option<u64>,
```
Comment (vusirikala) [inline crates/transaction-emitter-lib/src/args.rs:168]: What's the difference between `max_gas_per_txn` and `init_max_gas_per_txn`?

Comment (igor-aptos) [reply crates/transaction-emitter-lib/src/args.rs:168]: in transaction generator, there are set of transactions we use to initialize emitter, that we don't care about throughput (like creating worker accounts, publishing modules etc), but that need to go through - no matter what

those are called "init" transactions, and we can set different gas price, and now different max_gas

File: crates/transaction-emitter-lib/src/cluster.rs (hunk 2 above)
Comment (igor-aptos) [reply crates/transaction-emitter-lib/src/cluster.rs:99]: If devnet gets upgraded, and I forget to update the value. 

or hit devnet endpoints with command for testnet

**Review by manudhundi — APPROVED** [review]

**Pull request merged**